source/agora/node/Registry.d

```diff
@@ -186,6 +190,27 @@ public class NameRegistry: NameRegistryAPI
         return RegistryPayloadData.init;
     }
 
+    /// Internal endpoint, mimics `getValidator` but forward the query if needed
+    public final const(RegistryPayloadData) getValidatorInternal (in PublicKey key)
+    {
+        TypedPayload payload = this.zones[ZoneIndex.Validator].get(key);
+        if (payload != TypedPayload.init)
+            return payload.payload;
+
+        // Only query upstream if we are not the primary server for the zone
+        if (this.zones[ZoneIndex.Validator].type == ZoneType.primary)
```

Comment:
Secondary also shouldn't do this, only caching should do. Secondary is expected to get records through AXFR.

Reply:
What if a secondary registry is also a validator ? I wanted to leave open the possibility of secondary registries which are validator to bypass the delays between AXFR to just query upstream as needed.

Reply:
I see, from DNS side this looks a little problematic; secondary and primary can have different zone data for some time.

Reply:
At first I thought that would only happen if the `registry_address` configured is different from the primary, but then I realized the SOA also need to get updated. So yes, I think I will disable it authoritative registries.